core/src/main/scala-2/caliban/schema/ArgBuilderDerivation.scala

```diff
@@ -40,14 +39,18 @@ trait CommonArgBuilderDerivation {
 
     private val required = params.collect { case (label, default) if default.isLeft => label }
 
+    private val isValueType =
+      ctx.isValueClass || (ctx.annotations.exists(_.isInstanceOf[GQLValueType]) && params.length == 1)
+
     override private[schema] val partial: PartialFunction[InputValue, Either[ExecutionError, T]] = {
       case InputValue.ObjectValue(fields) if required.forall(fields.contains) => fromFields(fields)
     }
 
     def build(input: InputValue): Either[ExecutionError, T] =
       input match {
         case InputValue.ObjectValue(fields) => fromFields(fields)
-        case value                          => ctx.constructMonadic(p => p.typeclass.build(value))
+        case value if isValueType           => ctx.constructMonadic(p => p.typeclass.build(value))
```

Comment:
@ghostdogpr do you recon we should change the line above to `case InputValue.ObjectValue(fields) if !isValueType` as well?

Reply:
Do you mean the line below? I think it's okay to fail 🤔

Reply:
No I mean the line above (i.e., `case InputValue.ObjectValue(fields) => fromFields(fields)`). I'm wondering whether it should be changed to:

```scala
case InputValue.ObjectValue(fields) if !isValueType => fromFields(fields)
```

A case where this can be "problematic" is this:

```scala
        @GQLValueType
        case class Foo(value: Option[String])
        val ab = ArgBuilder.gen[Foo]
        assertTrue(
          ab.build(ObjectValue(Map())).isLeft
        )
```
Currently this test fails because `Map()` is a valid value, but it shouldn't be

Reply:
Yeah I think it makes sense to change as you suggested.